42/README.md

```diff
@@ -0,0 +1,85 @@
+
+
+### Type 2 scenarios
+
+These scenarios don't actually impact the sanctity of the blockchain but rather the shape of the data. In order to handle such scenarios following two approaches can used,
+
+#### Approach 1
+
+Create migration scripts for each new upgrade i.e. with each new release which results in a change in the shape of the data we create corresponding migrations scripts. Needless to say that writing such scripts might be tedious and in the worst case users can only incrementally upgrade to newer version
+
+#### Approach 2
+
+Tendermint allows to create follower nodes which don't validate transactions but rather just receive them and record them. We use this migrate data in the following manner,
+
+- The node operator create an new node which follows its existing validator node. 
+- This will allow this new follower node to replay all the blocks but store them according to the new data shape. 
+- Once this new follower node fully syncs with the current validator node the node operator shuts down Tendermint and changes the `prox_app` address in `config.toml` to the follower nodes BigchainDB server.
+- The operator can then go ahead and discard the existing older BigchainDB server and the corresponding data.
+
+The above process ensure the node keeps validating and accepting new blocks while the new database is being populated. The downtime experienced is when the Tendermint is shutdown to change the `proxy_app` address.
+
+
+### Type 3 scenarios
+When Tendermint migrates to a new version wherein new validation rules are not compatible with the existing blockchain it induces a hard fork. Tendermint encourages its users to create a new blockchain and achieve the current blockchain at particular agreed upon (by the network) height. In order to support such changes BigchainDB would also need to achieve the existing blockchain because the alternative approach is to run two Tendermint processes, one with the older validation logic and another with the newer validation logic. Since the latter approach is a huge hassle for BigchainDB users it would be best to recommend our user to achieve the blockchain.
+
+Following describes how to archive and start a new blockchain,
+- Create a TEP which proposes a block height `h` at which the current blockchain will stop creating any new blocks.
+- Once the TEP concludes and the height `h` has been chosen, validators stop proposing new blocks once height `h` has been reached.
+- At height `h` the merkle root of the UTXO of the current blockchain is provided as the genesis state of the new blockchain.
+- The validator set at height `h` should be used as the validator set in the genesis of the new blockchain.
+- Since from Tendermint's point of view this is a new blockchain, block height will start from `0` which implies that BigchainDB would need to achieve the existing blocks collection. All other collections would remain as is.
```

Comment:
NOTE: It is not confirmed if Tendermint will acknowledge the new blockchain as a fork and start from height `h` rather than height `0`. In case Tendermint doesn't acknowledge then BigchainDB will have to maintain the block heights such that forking at height `h` is accounted for.